sys/net/application_layer/dhcpv6/client.c

```diff
@@ -658,8 +658,8 @@ static int _preparse_advertise(uint8_t *adv, size_t len, uint8_t **buf)
         }
     }
     if ((cid == NULL) || (sid == NULL) ||
-        (IS_USED(MODULE_DHCPV6_CLIENT_IA_PD) && (ia_pd == NULL)) ||
-        (IS_USED(MODULE_DHCPV6_CLIENT_IA_NA) && (ia_na == NULL))) {
+        ((IS_USED(MODULE_DHCPV6_CLIENT_IA_PD) && (ia_pd == NULL)) &&
+        (IS_USED(MODULE_DHCPV6_CLIENT_IA_NA) && (ia_na == NULL)))) {
```

Comment:
Hmm, wouldn't this change cause the client to not ignore Advertisments at all if only `MODULE_DHCPV6_CLIENT_IA_NA` is being used? Maybe the check `(IS_USED(MODULE_DHCPV6_CLIENT_IA_PD) && (ia_pd == NULL)` should be removed instead?

Reply:
What if the `dhcpv6_client_ia_na` module is used but no interface has the `GNRC_NETIF_AAC_DHCP` flag set (yet)? Should this be an error?

Reply:
Should we set the initial value of `pref_val` to -1 then?

Reply:
> Should we set the initial value of `pref_val` to -1 then?

What is the connection there?

Reply:
Ah, nevermind, I thought you wanted to whole if statement